### PR TITLE
Allow substraction resulting in timestamp

### LINKED ISSUE #115: Datetime subtraction
I can't subtract two datetimes from each other. I can do this in pandas without issues. See this notebook: https://databricks-prod-cloudfront.cloud.databricks.com/public/793177bc53e528530b06c78a4fa0e086/0/12638605/100107/latest.html

Tony would like to show off this feature in his blog.

Screen Shot 2019-04-18 at 9 32 36 AM

### PR DESCRIPTION
**Pull request by HyukjinKwon**
Resolves #115

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @rxin

File: databricks/koalas/tests/test_series.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import unittest
+
+import pandas as pd
+
+from databricks import koalas
+from databricks.koalas.testing.utils import ReusedSQLTestCase, TestUtils
+
+
+class SeriesTest(ReusedSQLTestCase, TestUtils):
```
Comment (rxin) [inline databricks/koalas/tests/test_series.py]: maybe create a test_series_datetime, and we will need to put more stuff here.

**Comment by rxin** [comment]
Looks great.

@brookewenig  I think this is what you wanted.

File: databricks/koalas/series.py [hunk 2]
```diff
@@ -65,7 +65,17 @@ def _init_from_spark(self, scol, kdf, metadata=None, *args):
     # arithmetic operators
     __neg__ = _column_op(spark.Column.__neg__)
     __add__ = _column_op(spark.Column.__add__)
-    __sub__ = _column_op(spark.Column.__sub__)
+
+    def __sub__(self, other):
```
Comment (rxin) [inline databricks/koalas/series.py]: document that this one supports timestamp ...

File: databricks/koalas/series.py [hunk 3]
```diff
@@ -65,7 +65,19 @@ def _init_from_spark(self, scol, kdf, metadata=None, *args):
...
+        assert isinstance(other, Series), "subtraction should only be applied to koalas.Series."
```
Comment (rxin) [inline databricks/koalas/series.py]: do we not allow __sub__ a literal value? 

e.g. 

```
df['x'] - 1
```

File: databricks/koalas/series.py [hunk 4]
```diff
@@ -65,7 +65,19 @@ def _init_from_spark(self, scol, kdf, metadata=None, *args):
...
+
+        # Note that timestamp subtraction casts arguments to integer. This is to mimic Pandas's
+        # behaviours. Pandas returns 'timedelta64[ns]' from 'datetime64[ns]'s subtraction.
+        if isinstance(self.spark_type, TimestampType):
+            if not isinstance(other.spark_type, TimestampType):
+                raise TypeError('datetime subtraction can only be applied to datetime series.')
+            return Series(self._scol.cast('int') - other._scol.cast('int'), self._kdf)
```
Comment (rxin) [inline databricks/koalas/series.py]: if astype already works, i'd prefer using astype so we minimize manual construction of Series.

Comment (HyukjinKwon) [reply databricks/koalas/series.py]: Looks it goes infinite loop .. :).

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**